### PR TITLE
Update the exception of format constraint in routes

### PR DESCRIPTION
**Pull request by ferdynton**
Per https://github.com/rails/rails/issues/20264
[ci skip]

### CODE REVIEW

File: guides/source/routing.md [hunk 1]
```diff
@@ -706,6 +706,8 @@ end
 
 NOTE: Request constraints work by calling a method on the [Request object](action_controller_overview.html#the-request-object) with the same name as the hash key and then compare the return value with the hash value. Therefore, constraint values should match the corresponding Request object method return type. For example: `constraints: { subdomain: 'api' }` will match an `api` subdomain as expected, however using a symbol `constraints: { subdomain: :api }` will not, because `request.subdomain` returns `'api'` as a String.
 
+NOTE: There is an exception for the `format` constraint: while it's a method on the Request object, it's also an implicit optional parameter on every path. Segment constraints take precedence and the `format` constraint is only applied as such when enforced through a hash. To solve this, you can [use a lambda](#advanced-constraints).
```
Comment (vipulnsward) [inline guides/source/routing.md]: would be good to have an example like above one `constraints: { subdomain: :api }`

Comment (Ferdy89) [inline guides/source/routing.md]: I tried to add an applicable example. Could you take another look and tell me if it looks clear enough?

File: guides/source/routing.md [hunk 2]
```diff
@@ -706,6 +706,8 @@ end
 
 NOTE: Request constraints work by calling a method on the [Request object](action_controller_overview.html#the-request-object) with the same name as the hash key and then compare the return value with the hash value. Therefore, constraint values should match the corresponding Request object method return type. For example: `constraints: { subdomain: 'api' }` will match an `api` subdomain as expected, however using a symbol `constraints: { subdomain: :api }` will not, because `request.subdomain` returns `'api'` as a String.
 
+NOTE: There is an exception for the `format` constraint: while it's a method on the Request object, it's also an implicit optional parameter on every path. Segment constraints take precedence and the `format` constraint is only applied as such when enforced through a hash. For example, `get 'foo', constraints: { format: 'json' }` will match `GET  /foo` because the format is optional by default. However, you can [use a lambda](#advanced-constraints) like in `get 'foo', constraints: lambda { |req| req.format == :json }` and the route will only match explicit json requests.
```
Comment (maclover7) [inline guides/source/routing.md]: Please capitalize JSON.

Comment (Ferdy89) [inline guides/source/routing.md]: Done

**Comment by maclover7** [comment]
One comment, and then this can be merged!

**Comment by maclover7** [comment]
:+1: Thank you for your contribution!

**Comment by maclover7** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**